src/plugins/intel_cpu/src/emitters/tpp/x64/jit_equation_emitter.cpp

```diff
@@ -0,0 +1,139 @@
+        return std::make_pair(static_cast<libxsmm_blasint>(*++subtensor.rbegin()),
+                              static_cast<libxsmm_blasint>(*subtensor.rbegin()));
+    };
+
+    m_equation_id = libxsmm_meqn_create();
+    const auto op_metadata = libxsmm_create_meqn_op_metadata(m_equation_id, -1);
+    const auto sing_attr = libxsmm_create_matrix_arg_attributes(LIBXSMM_MATRIX_ARG_TYPE_SINGULAR, LIBXSMM_MATRIX_ARG_SET_TYPE_NONE, 0, 0);
+    libxsmm_blasint M, N;
+    for (const auto& op_desc : eq_tpp->get_op_descs()) {
+        switch (op_desc.get_arity()) {
+            case tpp::op::OpDescTPP::ARITY::BINARY: {
+                auto flags = op_desc.get_flags();
+                libxsmm_meqn_push_back_binary_op(op_metadata, op_desc, exec_dtype, flags);
+                break;
+            } case tpp::op::OpDescTPP::ARITY::UNARY: {
+                libxsmm_meqn_push_back_unary_op(op_metadata, op_desc, exec_dtype, LIBXSMM_MELTW_FLAG_UNARY_NONE);
+                break;
+            } case tpp::op::OpDescTPP::ARITY::ZERO: {
+                const auto arg_idx = static_cast<int>(op_desc);
+                std::tie(M, N) = get_MN(arg_idx);
+                auto metadata = libxsmm_create_meqn_arg_metadata(m_equation_id, arg_idx);
+                auto shape = libxsmm_create_meqn_arg_shape(N, M,
+                                                           static_cast<libxsmm_blasint>(io_strides[arg_idx]),
+                                                           io_dtypes[arg_idx]);
+                OV_CPU_JIT_EMITTER_ASSERT(libxsmm_meqn_push_back_arg(metadata, shape, sing_attr) == 0,
+                                          "Failed to push back arg to tpp equation");
+                break;
+            }
+            default:
+                OV_CPU_JIT_EMITTER_THROW("Unhandled tpp::op::OpDescTPP::ARITY");
+        }
+    }
+    // Note: for debug purposes it might be useful to serialize the equations graph here
+    // libxsmm_meqn_tree_print(m_equation_id);
+    // libxsmm_meqn_rpn_print(m_equation_id);
+    std::tie(M, N) = get_MN(static_cast<int>(io_port_descriptors.size()) - 1);
+    m_out_shape = libxsmm_create_meqn_arg_shape(N, M,
+                                                static_cast<libxsmm_blasint>(io_strides.back()),
+                                                io_dtypes.back());
+}
+
+size_t EquationTppEmitter::get_inputs_num() const {
+    return m_num_inputs;
+}
+
+const uintptr_t EquationTppEmitter::get_compiled_kernel_ptr() const {
+    return COMPILE_TPP_KERNEL(libxsmm_dispatch_meqn(m_equation_id, m_out_shape));
+}
+
+std::set<std::vector<element::Type>> EquationTppEmitter::get_supported_precisions(const std::shared_ptr<ov::Node>& node) {
+    // Note: TPPs have build-in convert semantics, so the equations should support any input precision (specified when created)
+    OV_CPU_JIT_EMITTER_ASSERT(ov::is_type<tpp::op::EquationTPP>(node), "Invalid node type");
```

Comment:
`node` might be nullptr (this is a default value). Can we add a check to avoid segfaults in such cases?

Reply:
Applied, thanks